kernel/posix/pthread_semaphore.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <pthread.h>
+#include <errno.h>
+
+#ifndef _PTHREAD_SEMAPHORE_H
+#define _PTHREAD_SEMAPHORE_H
+
+int sem_destroy(sem_t *semaphore)
+{
+	if (k_sem_count_get(semaphore->sem) != 0) {
```

Comment:
There's no need to get the count of a semaphore here: `EINVAL` is only signaled if `sem` is invalid.